### PR TITLE
feat: findObjExpression to match more cases

### PR DESCRIPTION
**Pull request by ig-ra**
handle more cases in findObjExpression

- rename findAttrObjExpression to findObjExpression
- handle two more cases in addition to `attribute` node case

### CODE REVIEW

File: pkg/analysis/completion.go [hunk 1]
```diff
@@ -306,53 +306,69 @@ func (a *Analyzer) keywordArgSymbols(fn query.Signature, args callWithArguments)
...
 			}
 		}
+	}
 
-		if expr != nil {
-			a.logger.Debug("dot completion",
-				zap.String("dot", dot.String()),
-				zap.String("expr", expr.String()))
-			return expr
+	// if not found, check nodes themselves
```
Comment (daabr) [inline pkg/analysis/completion.go:356]: Nit: please use full sentences as comments, i.e. use capitalization and punctuation: "If ... themselves."

File: pkg/analysis/completion.go [hunk 2]
```diff
@@ -306,53 +306,69 @@ func (a *Analyzer) keywordArgSymbols(fn query.Signature, args callWithArguments)
...
+	if dot == nil && expr == nil {
+		for i := len(nodes) - 1; i >= 0; i-- {
+			if nodeComparisonFunc(nodes[i]) == 0 && i > 0 {
```
Comment (daabr) [inline pkg/analysis/completion.go]:
```suggestion
		for i := len(nodes) - 1; i > 0; i-- {
			if nodeComparisonFunc(nodes[i]) == 0 {
```

File: pkg/analysis/completion_test.go [hunk 3]
```diff
@@ -336,3 +338,77 @@ s.i`, line: 1, char: 3, expected: []string{"items"}},
...
+			assert.ElementsMatch(t, tt.expectedNodeTypes, nodeTypes)
+
+			objNode := f.a.findObjectExpression(nodes, pt)
+			assert.Equal(t, objNode.Type(), "identifier")
+			assert.Equal(t, doc.Content(objNode), "foo")
+		})
+	}
+}
+
+func PrintNodeTree(d document.Document, n *sitter.Node, indent string) string {
```
Comment (daabr) [inline pkg/analysis/completion_test.go]: Please rename to `printNodeTree` so as not to expose a test helper.

**Review by daabr — APPROVED** [review]
Approved with a few nits

File: pkg/analysis/completion_test.go (hunk 3 above)
Comment (ig-ra) [reply pkg/analysis/completion_test.go]: sure

File: pkg/analysis/completion.go (hunk 2 above)
Comment (ig-ra) [reply pkg/analysis/completion.go]: done

File: pkg/analysis/completion.go (hunk 1 above)
Comment (ig-ra) [reply pkg/analysis/completion.go:356]: https://google.github.io/styleguide/go/decisions.html#:~:text=Comments%20that%20are%20complete%20sentences,if%20it%20is%20otherwise%20clear.

this comment isn't intended to be a full sentnence. 
on the other hand, comments which are full sentences are properly capitalized

**Pull request merged**